lib/super_good/solidus_taxjar/api.rb

```diff
@@ -53,7 +53,9 @@ def delete_transaction_for(order)
       end
 
       def show_latest_transaction_for(order)
-        taxjar_client.show_order order.number
+        latest_transaction_id =
+          OrderTransaction.latest_for(order)&.transaction_id || order.number
```

Comment:
We should only be aware of transactions that have an `OrderTransaction` associated with them.

If an order has an existing transaction, manually entered by other means external to our extension, there's no way of knowing what it's `transaction_id` would be. i.e. I don't think defaulting to `order.number` is a useful default here.

Reply:
The first `OrderTransaction` for an order will always be the `order.number` anyways.

Reply:
I don't think the fallback is safe, because we haven't considered what we what to do with orders that existed in the app before taxjar.

For example, let's say the following events happen.

1. Someone installs this gem (pre the reporting feature release)
2. Someone then uses this gem to calculate taxes. They then **manually** create transactions in the taxjar web app that correspond to the order.
    * The naming schema they used when manually creating the transactions is completely unknown to us. i.e. `MY-COMPANY-NAME-SOLIDUS-ORDER-NUMBER-28371-RANDOM-CHARACTERS`. We have no guarantee it will be the `order.number`
3. They then update this gem and **enable reporting**, and **don't backfill historical spree orders with their manually entered transactions**
4. They try to refund the above order
    * In that case, this fallback will fail, as we've assumed a default transaction naming schema that is wrong.
    
    
This is a larger problem, that affects all our of reporting API's when they are dealing with pre-existing orders. I've created a issue placeholder to pull out this larger discussion from the scope of this PR. https://github.com/SuperGoodSoft/solidus_taxjar/projects/1#card-71078848

I think for now, we should just throw, as we don't have a good way of handling this case, and we should just move this PR through and figure out how to handle that edge case later.